torch/csrc/jit/codegen/cuda/contiguity.h

```diff
@@ -39,14 +39,24 @@ class ContigIDs : public OptInDispatch {
   //! analyzed, in which case producer-to-consumer maps should be
   //! passed.
   //!
+  //! If ignore_indexability and ignore_halo_constraint are true,
+  //! ignore the constraint on indexing and halo, respectively. It is
+  //! the caller that is responsible for its correctness.
+  //!
+  //! The function interface with many parameters looks ugly, but it
+  //! is also important to make ignore_indexability and
+  //! ignore_halo_constraint explicit to avoid any surprise.
+  //!
   //! Not really sure why but clang-tidy only complains about
   //! std::unordered_map if passed as a const reference.
   ContigIDs(
```

Comment:
Just curious, how are the predicates generated for vectorization on merged contig-domains? I'm assuming it couldn't be one predicate for each root domain id anymore since some valid vector loads can be crossing the dimensions.

Reply:
Right, contig merging means predicates are generated at merge output domains, not original root domains.